keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -649,89 +629,182 @@ endpoints:
+- `ExternalName`: It doesn't make sense to export an `ExternalName` service.
+  They can't be merged with other exports, and it seems like it would only
+  complicate deployments by even attempting to stretch them across clusters.
+  Instead, regular `ExternalName` type `Services` should be created in each
+  cluster individually. If a `ServiceExport` is created for an `ExternalName`
+  service, an `InvalidService` condition will be set on the export.
+
+#### SuperclusterIP
+
+A non-headless `ServiceImport` is expected to have an associated IP address, the
+supercluster IP, which may be accessed from within an importing cluster. This IP
+may be a single IP used supercluster-wide or assigned on a per-cluster basis,
+but is expected to be consistent for the life of a `ServiceImport` from the
+perspective of the importing cluster. Requests to this IP from within a cluster
+will route to backends for the aggregated Service.
+
+Note: this doc does not discuss `NetworkPolicy`, which cannot currently be used
+to describe a selector based policy that applies to a multi-cluster service.
+
+#### DNS
+
+_Optional, but recommended._
+
+MCS aims to align with the existing [service DNS
+spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md). This
+section assumes familiarity with in-cluster Service DNS behavior.
+```
+<<[UNRESOLVED]>>
+Full DNS spec needed prior to Beta graduation following:
+https://github.com/kubernetes/dns/blob/master/docs/specification.md
+<<[UNRESOLVED]>>
+```
+When a `ServiceExport` is created, this will cause a domain name for the
+multi-cluster service to become accessible from within the supercluster. The
+domain name will be `<service>.<ns>.svc.supercluster.local`. 
+
+**SuperclusterIP services:** Requests to this domain name from within an importing
+cluster will resolve to the supercluster IP, which points to endpoints for pods
+within the underlying `Service`(s) across the supercluster.
+
+**Headless services:** Within an importing cluster, the supercluster domain name
+will have multiple `A`/`AAAA` records, each containing the address of a ready
+endpoint of the headless service. `<service>.<ns>.svc.supercluster.local` will
+resolve to the set of all ready pod IPs for the service.
+
+Pods backing a supercluster service may be addressed individually using the
+`<hostname>.<clustername>.<svc>.<ns>.svc.supercluster.local` format. Necessary
+records will be created based on each ready endpoint's hostname and the
+`multicluster.kubernetes.io/source-cluster` label on the `EndpointSlice`. This
+allows naming collisions to be avoided for headless services backed by identical
+`StatefulSets` deployed in multiple clusters.
+
+_Note: the total length of a FQDN is limited to 253 characters. Each label is
+independently limited to 63 characters, so users must choose host/cluster/service
+names to avoid hitting this upper bound._
+
+All service consumers must use the `*.svc.supercluster.local` name to enable
```

Comment:
Are we asserting that the name we write here is the only valid one, or is it configurable?

Reply:
We had discussed not allowing configuration, at least for now. There's nothing stopping a user/implementation from adding their own aliases but I don't think we need to make explicit statements about that

Reply:
The name suffix is sort of a handshake-agreement between the clients and the DNS impl.  I don't see why we wouldn't grant this the same flexibility that we grant cluster DNS.  Though again - maybe it is easier to loosen that or add aliases later.  Can you clarify the intent?  Then I am happy

Reply:
Added clarification